src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8453,20 +8454,17 @@ def align(
             ErrorMessage.not_implemented(
                 "Snowpark pandas 'align' method doesn't support 'fill_value'"
             )
-        if axis != 0:
-            # TODO: SNOW-1752856
-            ErrorMessage.not_implemented(
-                f"Snowpark pandas 'align' method doesn't support 'axis={axis}'"
-            )
+
         frame = self._modin_frame
         other_frame = other._query_compiler._modin_frame
 
-        if self.is_multiindex(axis=axis) or other._query_compiler.is_multiindex(
-            axis=axis
-        ):
-            raise NotImplementedError(
-                "Snowpark pandas doesn't support `align` with MultiIndex"
-            )
+        if axis is not None:
+            if self.is_multiindex(axis=axis) or other._query_compiler.is_multiindex(
```

Comment:
Should we raise this error when `axis=None` as well?

Reply:
+1. Also, do we have tests for multi index with `axis=None`?

Reply:
frame.is_multiindex has a check in [`num_index_levels`](https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/modin/plugin/_internal/frame.py#L482) that asserts axis = 0 or 1.  I'm not sure if this is intended or not, but I would run into the ValueError when axis=None.

Reply:
@sfc-gh-helmeleegy Oh not yet, I'll add the neg multiindex tests with axis 1 and None